test/http/http.ml

```diff
@@ -0,0 +1,48 @@
+open Stdune
+
+module Server = struct
+  type t =
+    { sock : Unix.file_descr
+    ; addr : Unix.sockaddr
+    }
+
+  let make addr =
+    let sock = Unix.socket ~cloexec:true Unix.PF_INET Unix.SOCK_STREAM 0 in
+    { sock; addr }
+  ;;
+
+  let port t =
+    match Unix.getsockname t.sock with
+    | Unix.ADDR_INET (_, port) -> port
+    | ADDR_UNIX _ -> failwith "no port defined"
+  ;;
+
+  let serve_once t =
```

Comment:
Is this function actually limited to serving only once? I thought the reason why the server only responds to a single request is that it doesn't call `Unix.accept` in a loop.

Reply:
That's confusing because it doesn't actually call `Unix.accept` and this module already had a function `accept` which does call `Unix.accept`.  This function sets up the socket but doesn't wait for a request (`Unix.accept` waits for a request).

Reply:
Indeed I renamed the old `accept` to be `start`.